saleor/plugins/webhook/tasks.py

```diff
@@ -82,17 +91,101 @@ def _get_webhooks_for_event(event_type, webhooks=None):
     return webhooks
 
 
-def trigger_webhooks_async(data, event_type, webhooks):
-    payload = EventPayload.objects.create(payload=data)
-    deliveries = create_event_delivery_list_for_webhooks(
-        webhooks=webhooks,
-        event_payload=payload,
-        event_type=event_type,
-    )
+def create_deliveries_for_subscriptions(
+    event_type, subscribable_object, webhooks, meta=None
+) -> List[EventDelivery]:
+    """Create webhook payload based on subscription query.
+
+    It uses a defined subscription query, defined for webhook to explicitly determine
+    what fields should be included in the payload.
+
+    :param event_type: event type which should be triggered.
+    :param subscribable_object: subscribable object to process via subscription query.
+    :return: List of event deliveries to send via webhook tasks.
+    """
+    if event_type not in SUBSCRIBABLE_EVENTS:
+        logger.info(
+            "Skipping subscription webhook. Event %s is not subscribable.", event_type
+        )
+        return []
+
+    context = initialize_context()
+    event_payloads = []
+    event_deliveries = []
+    for webhook in webhooks:
+        data = generate_payload_from_subscription(
+            event_type=event_type,
+            subscribable_object=subscribable_object,
+            subscription_query=webhook.subscription_query,
+            context=context,
+            app=webhook.app,
+        )
+        if not data:
+            # FIXME build better way of handling such cases
+            continue
+        if len(data) == 1 and "__typename" in data:
+            continue
```

Comment:
Maybe a line comment here why we add this `if`, to explain what case is this.

Reply:
after some investigation and consultations with @korycins we decided to add some tests and drop it as no longer viable

Reply:
Yes, I think that this is something that I left accidentally after cleaning up the implementation of POC